docs/decisions/0002-auto-enrollment-post-request-approval.rst

```diff
@@ -0,0 +1,158 @@
+  When the learner clicks the link, they will be lead to the course about page where they can view their course.
+  It's important for this email to be generic because it allows enterprise-access to be decoupled from the enrollment process
+  by not having to wait for enrollment and continue to focus on subsidy requests.
+  It doesn't care if a learner was enrolled or not, just that a subsidy was granted successfuly.
+* enterprise-access will consume subsidy-assigned events from license-manager and ecommerce.
+  The events might look like these:
+  ```
+    {
+        'event_type': 'org.openedx.ecommerce.coupon-code.assigned.v1',
+        'code': 'ABC123',
+        'coupon': 3,
+        'course_id': 'edX+DemoX',
+    },
+    {
+        'event_type': 'org.openedx.license-manager.license.assigned.v1',
+        'subscription_uuid': 'subscription-uuid',
+        'license': 'license-uuid',
+        'course_id': 'edX+DemoX',
+    }
+  ```
+* enterprise-access will send the now generic approval email to learners upon consumption of these events and update subsidy request states
+
+Consequences
+------------
+* A learner could get to the learner portal before being auto enrolled, but it's an edge case and they should still be able to enroll manually.
+* Enrollment could fail for a learner but they wouldn't know, they should still be able to enroll manually.
+* We don't have to consume enrollment events from the LMS, removing a step in our architecture and reducing complexity.
+* enterprise-access stays within its bounded context.
+
+
+Alternatives considered
+=======================
+
+Alternative Solution 1
+======================
+
+Slightly tweaking the solution above, ecommerce and license-manager would send the subsidy assigned event `after` enrolling a learner.
+That would allow enterprise-access to send variants of the approval email and also guarantee that a learner is enrolled by the time the email is sent.
+
+The subsidy assigned event might look something like this:
+```
+  {
+      'event_type': 'org.openedx.ecommerce.coupon-code.assigned.v1',
+      'code': 'ABC123',
+      'coupon': 3,
+      'is_redeemed': True
+  }
+```
+
+Consequences
+------------
+* We would want to send a subsidy assigned event even if enrollment fails.
+
+
+Alternative Solution 2
+======================
+
+enterprise-access still calls ecommerce and license-manager to assign a subsidy, then sends a request approved event
+to the event bus. The subsidy granting services will consume the events, redeem the granted subsidy,
+and enroll a learner in the course specified in the request.
+
+Changes required
+----------------
+* We will no longer send an email to the learner at the time of approval. Instead, we will send an email to
+  the learner when they have been enrolled.
+* enterprise-access will consume enrollment events from the LMS, check if the enrollment corresponds to an approved
```

Comment:
Line 60 above states:
>  enterprise-access consumes the subsidy assigned event and changes the state on the subsidy request to `fulfilled`.
   It then sends an request approved email to the learner of the request. As far as enterprise-access knows, it's work is done at this point.

Will these events be consumed or is that out of scope?